se/se_epub.py

```diff
@@ -508,6 +508,28 @@ def recompose(self, output_xhtml5: bool, extra_css_file: Path = None) -> str:
 
 			file_css = regex.sub(r"\s*@(charset|namespace).+?;\s*", "\n", file_css).strip()
 
+			# Convert background-image URLs (from local.css only) to base64
+			if filename == "local.css":
+				convert_css = file_css
+				for img in regex.finditer(pattern=r'(url\(".*\.)(svg)("\))', string=convert_css):
+					src = img.group().replace("../", "")
+					img_file = regex.sub(r'.*"(.*)".*', r'\1', src)
+					with open(self.content_path / img_file, "rb") as binary_file:
+						image_contents_base64 = base64.b64encode(binary_file.read()).decode()
+
+					if img_file.endswith(".svg"):
+						replacement = f"url('data:image/svg+xml;base64,{image_contents_base64}')"
+					elif img_file.endswith(".jpg"):
+						replacement = f"url('data:image/jpg;base64,{image_contents_base64}')"
+					elif img_file.endswith(".png"):
+						replacement = f"url('data:image/png;base64,{image_contents_base64}')"
+					else:
+						replacement = src
+
+					convert_css = f"{convert_css[:img.start()]}{replacement}{convert_css[img.end():]}"
```

Comment:
This is clever but why not just do a straight string replace? Something like `file_css = file_css.replace(<full regex match>, replacement)`?

Reply:
Mostly because my experience with the python regex library is the past few days of reading about it - that looks like a much better way. I'll update to fix things based on all three comments and send it over.

Reply:
new commit created. I ran this locally and it looked good, did the recompose successfully.